src/frontend/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,68 @@
+      <small id="usernameError" class="form-text text-danger" *ngIf="registrationForm.controls.username.errors.required">username is required.</small>
+      <small id="usernameError" class="form-text text-danger" *ngIf="registrationForm.controls.username.errors.minlength || registrationForm.controls.username.errors.maxlength">username must be between 5 to 9 characters.</small>
+      </div>
+    </div>
+
+    <!--<div class="form-group">-->
+      <!--<label for="inputEmail">Email</label>-->
+      <!--<input type="email" class="form-control w-50" id="inputEmail" aria-describedby="emailHelp"-->
+             <!--placeholder="Enter email" [(ngModel)]="userDto.email" name="email" required>-->
+      <!--<small id="emailHelp" class="form-text text-muted">provide email to sent you verification code.</small>-->
+    <!--</div>-->
+
+    <div class="form-group">
+      <label for="inputPassword">Password</label>
+      <input type="password" class="form-control w-50" id="inputPassword" aria-describedby="passwordHelp"
+             [(ngModel)]="userDto.password" formControlName="password">
+      <small id="passwordHelp" class="form-text text-muted">Password must consist of at least one big character, one number and has to be 8 character's long.</small>
+      <div *ngIf="registrationForm.controls.password.invalid && (registrationForm.controls.password.touched || registrationForm.controls.password.dirty)">
+        <small id="passwordRequired" class="form-text text-danger" *ngIf="registrationForm.controls.password.errors.required">password is required.</small>
+        <small id="passwordMinLength" class="form-text text-danger" *ngIf="registrationForm.controls.password.errors.minlength">password must be at least 8 characters long.</small>
+        <small id="passwordPattern" class="form-text text-danger" *ngIf="registrationForm.controls.password.errors.pattern">password must have at least one uppercase and lowercase character and one digit.</small>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="inputPasswordMatchup">Re-Password</label>
+      <input type="password" class="form-control w-50" id="inputPasswordMatchup" aria-describedby="passwordMatchupHelp" required
+             minlength="8" formControlName="repassword">
+      <small id="passwordMatchupHelp" class="form-text text-muted">Password must match with the one above.</small>
+      <!--<div *ngIf="registrationForm.controls.repassword.invalid && (registrationForm.controls.repassword.touched || registrationForm.controls.repassword.dirty)">-->
+      <div *ngIf="registrationForm.controls.repassword.invalid && (registrationForm.controls.repassword.touched || registrationForm.controls.repassword.dirty)">
+        <small id="passwordError" class="form-text text-danger" *ngIf=" registrationForm.controls.repassword.errors.passwordConfirming
```

Comment:
ten if musi się znajdować w kontrolerze czyli w komponencie, stwórz na to jakąś funkcję, a tu tylko ngIf=i nazwa funkcji